### PR TITLE
Allow users with "manage_service" permission to edit team members' email addresses

### PR DESCRIPTION
**Pull request by CrystalPea**
As per: https://www.pivotaltracker.com/story/show/163561738

#### Manage team members page: 
- link text changed
screen shot 2019-02-20 at 17 36 39

#### Edit team member page:
- add link to change team member's email address
screen shot 2019-02-21 at 10 43 17

#### Change team member's email address:
- whole new page
screen shot 2019-02-20 at 17 37 06

#### Confirm email address change:
- whole new page
screen shot 2019-02-22 at 09 46 42

### CODE REVIEW

File: app/templates/views/edit-user-permissions.html [hunk 1]
```diff
@@ -14,7 +14,7 @@ <h1 class="heading-large">
   </h1>
 
   <p>
-    {{ user.email_address }}
+    {{ user.email_address }} <a href="{{ url_for('.edit_user_email', service_id=current_service.id, user_id=user.id)}}">Change</a>
```
Comment (quis) [inline app/templates/views/edit-user-permissions.html]: Putting a little bit more space between the email address and the link will look nicer (an em-space is a space the width of the letter ‘M’):
```suggestion
    {{ user.email_address }}&emsp;<a href="{{ url_for('.edit_user_email', service_id=current_service.id, user_id=user.id)}}">Change</a>
```

Comment (CrystalPea) [reply app/templates/views/edit-user-permissions.html]: cool 👍 . Like this, aye?

screen shot 2019-02-21 at 10 43 17

File: app/templates/views/manage-users/confirm-edit-user-email.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+{% block maincolumn_content %}
+
+<h1 class="heading-large">Confirm change of email address</h1>
+
+<div class="grid-row">
+  <div class="column-whole">
+  {% call form_wrapper() %}
+    <p>New email address:</p>
+    <div class="panel panel-border-wide">
```
[the remark this answers is not in the document]
Comment (CrystalPea) [reply app/templates/views/manage-users/confirm-edit-user-email.html]: Good call, thank you and done 👍 :)

File: app/main/views/manage_users.py [hunk 3]
```diff
@@ -122,6 +135,60 @@ def remove_user_from_service(service_id, user_id):
...
+        service_id=service_id
+    )
+
+
+@main.route("/services/<service_id>/users/<user_id>/edit-email/confirm", methods=['GET', 'POST'])
+@login_required
+@user_has_permissions('manage_service')
+def confirm_edit_user_email(service_id, user_id):
+    user = user_api_client.get_user(user_id)
+    new_email = session['team_member_email_change']
```
Comment (allait) [inline app/main/views/manage_users.py]: This should probably go back to the edit page if there's nothing in the session.

File: app/main/views/manage_users.py [hunk 4]
```diff
@@ -122,6 +135,60 @@ def remove_user_from_service(service_id, user_id):
...
+            user_api_client.update_user_attribute(user_id, email_address=new_email)
+        except HTTPError as e:
+            if e.status_code == 403:
+                flash("You don't have permission to edit users emails for this service", 'info')
+                return redirect(url_for(
+                    '.manage_users',
+                    service_id=service_id))
+            else:
+                abort(500, e)
+
```
Comment (allait) [inline app/main/views/manage_users.py]: We should clean up the session in a `finally` block here

File: app/main/views/manage_users.py (hunk 3 above)
Comment (CrystalPea) [reply app/main/views/manage_users.py]: Good call, I added code and test for that now 👍

File: app/main/views/manage_users.py (hunk 4 above)
Comment (CrystalPea) [reply app/main/views/manage_users.py]: Ahh, good call, I added that block now, I didn't know about it before :)

**Pull request merged**